dev-tools/mage/check.go

```diff
@@ -228,19 +229,22 @@ func CheckDashboardsFormat() error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to read dashboard file %s", file)
 		}
-		var dashboard Dashboard
-		err = json.Unmarshal(d, &dashboard)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse dashboard from %s", file)
-		}
 
-		module := moduleNameFromDashboard(file)
-		errs := dashboard.CheckFormat(module)
-		if len(errs) > 0 {
-			hasErrors = true
-			fmt.Printf(">> Dashboard format - %s:\n", file)
-			for _, err := range errs {
-				fmt.Println("  ", err)
+		r := bufio.NewReader(bytes.NewReader(d))
+		count := 0
+		for {
+			line, err := r.ReadBytes('\n')
+			count++
+			if err != nil {
+				if err == io.EOF {
+					if checkDashboardForErrors(file, line, count) {
```

Comment:
I think that decoding here using a json decoder would make the code cleaner, something like this:
```suggestion
		r := bufio.NewReader(bytes.NewReader(d))
		decoder := json.NewDecoder(r)
		count := 0
		for decoder.More() {
			var dashboard Dashboard
			err := decoder.Decode(&dashboard)
			... err handling ...
			count++
			err = dashboard.CheckFormat()
			....
```

Reply:
Cleaner? IMHO this is cleaner because `checkDashboardForErrors` hides everything from the caller and lets you know exactly what it is doing. I consider trying to decode the dashboard error checking, so I would keep it in `checkDashboardForErrors`.

Reply:
Well, maybe "cleaner" was not the best word :grimacing:, sorry. But wit a decoder we don't need to handle lower level things as delimiters or EOFs. This may give an answer too to https://github.com/elastic/beats/pull/27553#discussion_r694114126, `json.Decoder` can decode sequential JSON objects without any delimiter or delimited by any number of whitespaces or newlines.

Reply:
Btw, you have a good point about encapsulating all the checks, including decoding, inside the same function, but this could be achieved too if the check function receives a reader or a decoder as parameter.